pkg/util/resource/resource_manager_test.go

```diff
@@ -63,7 +63,9 @@ func TestResourceManager(t *testing.T) {
 		assert.Equal(t, "foo2", res.Get())
 		res = manager.Delete("test", "foo")
 		assert.Equal(t, "foo2", res.Get())
-		time.Sleep(time.Second)
+		for manager.Delete("test", "foo") != nil {
+			time.Sleep(500 * time.Millisecond)
+		}
```

Comment:
NIT: we could use testify `Eventually` function to perform this periodic check with timeout 
```suggestion
		assert.Eventually(t, func() bool {
			return manager.Delete("test", "foo") != nil
		}, time.Second*5, time.Millisecond*100)
```